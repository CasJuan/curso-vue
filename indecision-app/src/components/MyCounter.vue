<template>

    <section>
        <h3>Counter: {{ counter }}</h3>
        <h3 data-testid="square-label">Square: {{ squareCounter }}</h3>

        <div>
            <button class="p-5  bg-blue-500 rounded hover:bg-blue-700 mr-2" @click="counter++"> +1</button>
            <button class="p-5  bg-blue-500 rounded hover:bg-blue-700 mr-2" @click="counter--"> -1</button>
        </div>
    </section>


</template>


<script lang="ts" setup>
import { useCounter } from '../composables/useCounter';



/* import {computed, ref} from 'vue'; */

/* const props = defineProps({
    value: {type: Number, required:true},
    }); */
    
    interface Props {
        value: number;
    }
    const props = defineProps<Props>(); 
    
    const {counter, squareCounter} =  useCounter(props.value);
/* 
    const counter = ref(props.value);
    const squareCounter = computed(() => counter.value * counter.value); */

</script>

<!-- /* <style scoped lang="postcss">
    .btn{
        @apply p-5 bg bg-blue-500 rounded hover:bg-blue-700 mr-2;
    }

</style>  */ -->

