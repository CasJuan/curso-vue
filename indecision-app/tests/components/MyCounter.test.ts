import { describe, expect, test } from "vitest";
import MyCounter from '@/components/MyCounter.vue';
import { mount } from '@vue/test-utils'

describe('<MyCounter/>', () => {
    test ('shoul match snapshot', () => {
        
        const wrapper = mount(MyCounter, {
            props: {
                value: 5,
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    test('renders the counter value correctly', () => {
        const wrapper = mount(MyCounter, {
            props: {
                value: 5,
            }
        });
        
        const [counterLabel, squareLabel] = wrapper.findAll('h3');


        //console.log(expect(wrapper.find('h3').text()).toContain(`Counter:${value}`));
        //console.log(expect(wrapper.find('[data-testid="square-label"').text()).toContain(`Square:${value * value}`));


        expect(counterLabel.text()).toContain(`Counter: ${value}`);
        expect(squareLabel.text()).toContain(`Square: ${value * value}`);

        //console.log(wrapper.html());

    })

})