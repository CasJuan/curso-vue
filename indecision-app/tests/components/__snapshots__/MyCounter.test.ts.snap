// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter/> > shoul match snapshot 1`] = `
"<section>
  <h3>Counter: 5</h3>
  <h3 data-testid="square-label">Square: 25</h3>
  <div><button class="p-5 bg-blue-500 rounded hover:bg-blue-700 mr-2"> +1</button><button class="p-5 bg-blue-500 rounded hover:bg-blue-700 mr-2"> -1</button></div>
</section>"
`;
